self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -22,43 +22,378 @@
+     */
+    private ProjectInternal findOrCreateProject(final int projectId) {
+        ProjectInternal project = projects.stream()
+            .filter(p -> p.projectId() == projectId)
+            .findFirst()
+            .orElse(null);
+        if (project == null) {
+            project = new ProjectInternal(projectId, this);
+            projects.add(project);
+        }
+        return project;
+    }
+
+    /**
+     * Returns a contributor by id from the contributors set.
+     * If the contributor is not found, a contributor will
+     * be created and added to set.
+     *
+     * @param cbId Contributor's id
+     * @return Internal representation of a Contributor
+     */
+    private ContributorInternal findOrCreateContributor(final int cbId) {
+        ContributorInternal contributor = contributors.stream()
+            .filter(c -> c.contributorId() == cbId)
+            .findFirst()
+            .orElse(null);
+        if (contributor == null) {
+            contributor = new ContributorInternal(cbId, this);
+            contributors.add(contributor);
+        }
+        return contributor;
+    }
+
+
+    /**
+     * In memory implementation of a {@link Project}.
+     * <br>
+     * Aids {@link InMemoryContracts}
+     * <br>
+     * Has capability to link a contract.
+     * See: {@link ProjectInternal#addContract(Contract)}.
+     */
+    private static final class ProjectInternal implements Project {
+
+        /**
+         * Project's id.
+         */
+        private final int id;
+        /**
+         * Global contracts. Access to {@link InMemoryContracts }
+         * Grants access to any arbitrary project's contracts using
+         * {@link Contracts#ofProject(int) from any point of the object graph.
+         */
+        private final Contracts rootContracts;
+
+        /**
+         * Contracts associated to this project.
+         */
+        private final Set<Contract> ownedContracts = new HashSet<>();
+
+        /**
+         * Constructor.
+         *
+         * @param id Project id
+         * @param rootContracts Global contracts.
+         *                      Access to {@link InMemoryContracts }
+         */
+        ProjectInternal(final int id, final Contracts rootContracts) {
+            this.id = id;
+            this.rootContracts = rootContracts;
+        }
+
+        /**
+         * Adds a contract to ownedContracts.
+         *
+         * @param contract Contract to be added.
+         */
+        void addContract(final Contract contract) {
+            ownedContracts.add(contract);
+        }
+
+        @Override
+        public int projectId() {
+            return id;
+        }
+
+        @Override
+        public ProjectManager projectManager() {
+            return null;
+        }
+
+        @Override
+        public Repo repo() {
+            return null;
+        }
+
+        @Override
+        public Contracts contracts() {
+            return new Contracts() {
```

Comment:
@criske I think ``return this.rootContracts.ofProject(projectId)`` is enough here.

Reply:
we will get stack-overflow